core/src/test/java/org/apache/accumulo/fate/zookeeper/ZooLockTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.fate.zookeeper;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+public class ZooLockTest {
+
+  @Test
+  public void testSortAndFindLowestPrevPrefix() throws Exception {
+    List<String> children = new ArrayList<>();
+    children.add("zlock#00000000-0000-0000-0000-FFFFFFFFFFFF#0000000007");
+    children.add("zlock#00000000-0000-0000-0000-EEEEEEEEEEEE#00000000010");
```

Comment:
IIRC, @EdColeman told me this. I don't have anything to back it up.

Reply:
I could remove the length checks in the `sortChildrenByLockPrefix` but leave in the check for the `#` at index 42. This would allow for longer sequence numbers.

Reply:
The zookeeper Programmers guide specifies the length and format of the naming.
[ZooKeeper - Sequence Nodes - Unique Naming](https://zookeeper.apache.org/doc/r3.5.9/zookeeperProgrammers.html#Sequence+Nodes+--+Unique+Naming)
` The counter has a format of %010d -- that is 10 digits with 0 (zero) padding (the counter is formatted in this way to simplify sorting), i.e. "0000000001". `

Reply:
Thanks @EdColeman!

Reply:
I did take a quick look to see if there was a public definition of the length in the ZooKeeper code - but I didn't see anything - even with that, the length should be stable - alternatively we could just check that it only contains chars [0-9].  Thinking about the check - rather than length - Parsing the uuid with UUID.from(string) would be a stronger check.

Reply:
Thanks @EdColeman  for the background info.  My overall motivation for asking these questions is that I think its good for code that reads persisted data to be skeptical that the data is at it should be.  With that motivation in mind I have been wondering what sanity checks could be done for the lock nodes.

Reply:
I have a fix in progress, running tests now. Should be uploaded shortly.

Reply:
I merged the methods that gets the children and sorts them, into one [here](https://github.com/apache/accumulo/pull/1866/files#diff-d53a0e8d1a6246f0196696217ef667f8730e51b93ffdea614d380315094e2387R176). Based on this thread, we might want to relax or remove the restriction on the ephemeral sequence number size.